src/peer_conn_protocol.rs

```diff
@@ -24,48 +20,28 @@ pub struct NewPeerConnection {
     pub peer: Peer,
 }
 
-#[derive(Debug, Clone)]
-pub struct P2PHandshake {
-    pub conn_id: u64,
-    pub peer: Peer,
-}
-
-pub fn start_p2p_state(
-    receiver: Receiver<P2POperation>,
-    messenger: Sender<MessengerOperations>, 
+pub fn send_p2p_handshake(
+    conn_id: u64,
+    peer_address: String,
+    peer_port: u16,
+    messenger: Sender<MessengerOperations>,
+    receiver: Receiver<P2POperations>,
```

Comment:
What's the receiver for? It's already received the command by having this function be called